src/subcommand/wallet/transaction_builder.rs

```diff
@@ -1070,4 +1092,67 @@ mod tests {
       })
     )
   }
+
+  #[test]
+  fn send_inscriptions_with_custom_fee_rate() {
+    let utxos = vec![(outpoint(1), Amount::from_sat(10_000))];
+
+    pretty_assert_eq!(
+      TransactionBuilder::build_transaction(
+        satpoint(1, 0),
+        BTreeMap::from([(
+          satpoint(1, 0),
+          "bed200b55adcf20e359bbb762392d5106cafbafc48e55f77c94d3041de3521da"
+            .parse()
+            .unwrap()
+        )]),
+        utxos.into_iter().collect(),
+        recipient(),
+        vec![change(0), change(1)],
+        FeeRate::try_from(25.8).unwrap(),
+      ),
+      Ok(Transaction {
+        version: 1,
+        lock_time: PackedLockTime::ZERO,
+        input: vec![tx_in(outpoint(1))],
+        output: vec![tx_out(7_445, recipient())],
+      })
+    )
+  }
+
+  #[test]
+  fn send_inscriptions_with_custom_fee_rate_calculated() {
```

Comment:
I think we can keep this one and remove the other one, and we can call it `fn build_transaction_with…`